src/metabase/upload.clj

```diff
@@ -215,12 +216,25 @@
       (throw (ex-info (tru "Unable to recognise file separator")
                       {:status-code 422}))))
 
+(def ^:private allowed-extensions #{nil "csv" "tsv" "txt"})
```

Comment:
The frontend does not allow txt or no extension, but I'm worried that there are customers using the API directly, and these are still very reasonable cases I've encountered before.